projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/VectorType.java

```diff
@@ -33,15 +33,24 @@
 
 public class VectorType extends AggregateType {
 
-    private final PrimitiveType elementType;
+    private final Type elementType;
     private final int length;
 
     public VectorType(PrimitiveType elementType, int length) {
         this.elementType = elementType;
         this.length = length;
     }
 
-    public PrimitiveType getElementType() {
+    public VectorType(PointerType elementType, int length) {
+        this.elementType = elementType;
+        this.length = length;
+    }
+
+// public PrimitiveType getElementType() {
```

Comment:
Yes, I did and I already noticed this myself. Sorry about my unneccessary comment.